interfaces/systemd/backend.go

```diff
@@ -181,6 +187,35 @@ func deriveContent(spec *Specification, appSet *interfaces.SnapAppSet) map[strin
 	return content
 }
 
+func ensureSnapServiceOverrides(spec *Specification, appSet *interfaces.SnapAppSet) error {
+	overrides := make(map[string][]wrappers.SnapServiceOverride)
+	for name, override := range spec.snapServiceOverrides {
+		svcOverride := wrappers.SnapServiceOverride{
+			Name:    name,
+			Content: override.content,
+		}
+		overrides[override.plugOrSlot] = append(overrides[override.plugOrSlot], svcOverride)
+	}
+
+	svcOverrides := make(map[*snap.AppInfo][]wrappers.SnapServiceOverride)
+	for _, svc := range appSet.Info().Services() {
+		for _, plug := range svc.Plugs {
+			if plugOverrides, ok := overrides[plug.Name]; ok {
+				svcOverrides[svc] = append(svcOverrides[svc], plugOverrides...)
+			}
+		}
+		for _, slot := range svc.Slots {
+			if sloteOverrides, ok := overrides[slot.Name]; ok {
+				svcOverrides[svc] = append(svcOverrides[svc], sloteOverrides...)
+			}
+		}
+	}
+	opts := &wrappers.EnsureSnapServiceOverridesOptions{
+		Prefix: "70-snap.interface.",
```

Comment:
won't using just this as prefix mean that we touch/remove files of different interfaces even if nothing changes about them? what happens on connect?

Reply:
Good point, but actually internally `EnsureFileState` first checks if the target content matches the current content, so only drop-ins that are changed/removed will be updated. 

There is a bug here though, where the prefix should contain at least the snap name, something like `70-snap.<snap-name>.interface.`.